### ISSUE TITLE
Feature: Heuristics for choosing a SIMD variant over the corresponding scalar non-SIMD version

### ISSUE THREAD

**Issue by nordlow** [ashvardanian/StringZilla#184]
### Describe what you are looking for

For a given function `f`, are there any (typically architecture-specific) heuristics for when a SIMD version opposite to a serial version of an algorithm should be chosen?

For instance, for 

```c
sz_find(sz_cptr_t haystack, sz_size_t h_length, sz_cptr_t needle, sz_size_t n_length) {
```

one such heuristics could be

```c
bool useSIMD(size_t h_length, size_t n_length) {
    if (h_length > 1024) {
        return true;
    } else if (h_length / n_length >= 4) {
        return true;
    }
}
```

.

### Can you contribute to the implementation?

- [X] I can contribute

### Is your feature request specific to a certain interface?

It applies to everything

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by ashvardanian** [comment]
There are definitely spots where platform-specific performance tuning can be a great idea. This can be a topic for a proper research project, potentially exposing threshold variables with `extern` and tuning our `sz_find`, `sz_copy`, and other kernels for the target platform. It get's a bit trickier, if the same kernel depends on multiple variables and we are forced to grid-search...

Are you familiar with similar projects, @nordlow?

**Comment by nordlow** [comment]
Unfortunately not.

Thanks for the response anyway.

**Comment by ashvardanian** [comment]
There is now a `SZ_SWAR_THRESHOLD` that you can use for this purpose, @nordlow 🤗

**Comment by ashvardanian** [comment]
Yes, @nordlow, it's a shared constant, but not all functions use it. In some cases SIMD is universally better in real usage, so they don't even check it. I understand, this is a bit of a design limitation, but felt like a good balance of API verbosity and usability. I'm open to suggestions for the v5, if you have any.

**Comment by nordlow** [comment]
Ok. Thanks a lot for now.
